pkg/operation/botanist/botanist.go

```diff
@@ -150,6 +150,16 @@ func New(ctx context.Context, o *operation.Operation) (*Botanist, error) {
 	if err != nil {
 		return nil, err
 	}
+	//TODO(acumino): Need to add when node local dns is enabled
+	if o.Shoot.NodeLocalDNSEnabled {
+		o.Shoot.Components.SystemComponents.NodeLocalDNS, err = b.DefaultNodeLocalDNS()
+		if err != nil {
+			return nil, err
+		} else {
+			o.Shoot.Components.SystemComponents.NodeLocalDNS = nil
+		}
```

Comment:
I cannot follow this `else` block. If there was not error constructing the DefaultNodeLocalDNS, then set it to nil? Why? Did you rather wanted to set the NodeLocalDNS field to nil when node local dns is not enabled?